yarn-project/circuit-types/src/logs/l1_payload/l1_note_payload.ts

```diff
@@ -136,4 +147,26 @@ export class L1NotePayload extends L1Payload {
       this.noteTypeId.equals(other.noteTypeId)
     );
   }
+
+  static #getPublicValuesAndRemainingCipherText(input: Buffer): [Fr[], Buffer] {
+    const reader = BufferReader.asReader(input);
+    const numPubliclyDeliveredValues = reader.readUInt8();
+
+    const remainingData = reader.readToEnd();
+    const publicValuesData = remainingData.subarray(
+      remainingData.length - numPubliclyDeliveredValues * Fr.SIZE_IN_BYTES,
+      remainingData.length,
+    );
+    if (publicValuesData.length % Fr.SIZE_IN_BYTES !== 0) {
+      throw new Error(`Public values byte length is not a multiple of Fr size. Length: ${publicValuesData.length}`);
+    }
```

Comment:
I don't understand how this error could be triggered, given the way the array is constructed above

Reply:
It gets triggered in some of our tests because L2Block.random populates logs with random values.

Reply:
Fixed it in https://github.com/AztecProtocol/aztec-packages/pull/8538/commits/4ffc89f534f8b1ccd5e8d780701dd2376e500a0c